src/main/scala/eisner/EisnerPlugin.scala

```diff
@@ -59,18 +69,38 @@ object EisnerPlugin extends AutoPlugin with ReflectionSupport with SnippetSuppor
       Thread.currentThread.setContextClassLoader(classOf[PromiseException].getClassLoader)
 
       if (topologyDescriptions.nonEmpty) {
-        val fs = Future.traverse(topologyDescriptions) {
-          case (topologyName, topology) =>
-            topology.toSVG.map { svg =>
-              val f = new File(s"$cacheDir/${dotsToSlashes(topologyName)}.svg")
-              IO.write(f, svg)
-              f
-            }
+        val config = Config(eisnerColorSubtopology.value, eisnerColorTopic.value, eisnerColorSink.value)
+        val topologiesWithDots = topologyDescriptions
+          .map {
+            case (n, td) => (n, td, td.toDot(config))
+          }
+          .collect {
+            case (n, td, Right(d)) => (n, td, d)
+          }
```

Comment:
this double traversal is not ideal but we don't have real `File`s to check against to see if we can avoid regenerating the SVGs (which is expensive time-wise), we need to build them every time anyway, but we can limit ourselves to either the actual topology description or the `.dot` file resulting, i'm caching the latter since now our `.dot` files depend on `Context`